### DISCUSSION TITLE
Google Login - One-tap sign-up shows up on refresh even after successful login

### DISCUSSION THREAD

**Discussion by kumarjitsen** [abacritt/angularx-social-login#624] [General]
Hi,

Followed the steps to implement Google sign-in and sign-out ("@abacritt/angularx-social-login": "^1.2.3") and asl-google-signin-button. After I'm successfully logged-in, on refresh the "One-tap sign-up" modal shows up on the top-right asking to sign-in again!

`
user: SocialUser;

this.socialAuthService.authState.subscribe(async (authState) => {
      this.user = authState;
      if (authState) {
        await lastValueFrom(this.authService.login('Google', authState.idToken)).then((tokenData) => {
          this.sharedDataService.isLoggedIn.next(true);
        });
        this.isShowSpinner = false;
        this.router.navigate(['']);
      }
    });`

**Comment by buhal** [comment]
Hi I have the same problem and it turns out that you can remove this by adding some more parameters in the app.module.ts
You need to define:
`const googleLoginOptions: GoogleInitOptions = {
  oneTapEnabled: false, // default is true
};`
And then in the providers use the created GoogleInitOptions:
` {
      provide: 'SocialAuthServiceConfig',
      useValue: {
        autoLogin: true,
        providers: [
          {
            id: GoogleLoginProvider.PROVIDER_ID,
            provider: new GoogleLoginProvider(
              environment.PROVIDER_ID,
              googleLoginOptions
            ),
          },
        ],
        onError: (err) => {
          console.error(err);
        },
      } as SocialAuthServiceConfig,
    },`

**Reply by kumarjitsen** [reply]
Thanks @buhal 
While it does solve the problem, the underlying problem still exists (one I didn't mention, but was hoping will have the same root cause).

I have a logout button, which basically calls `this.socialAuthServive.signOut()` but on browser reload, this method fails with Error: "Not logged in" (which is basically coming from https://github.com/abacritt/angularx-social-login/blob/d4a0038f873085d66b6a62bfdf26be38739e38d8/projects/lib/src/socialauth.service.ts#L232) 

This means on refresh, the service loses the user variable, and there's no hook for me to set it on app-initializer (which ideally I shouldn't have to do). The issue is also there with the demo app https://abacritt.github.io/angularx-social-login/ (autoLogin works mostly, which is why it's not that evident). Not sure how to handle the signout issue.

**Reply by buhal** [reply]
Indeed, now when I test this it really throws this error when you refresh the browser and then signOut().
I hope someone will enlighten un on this problem too.

**Comment by karthik2265** [comment]
@buhal  This is the best npm package I have found so far for authorization in angular apps, hope the issue is resolved soon
